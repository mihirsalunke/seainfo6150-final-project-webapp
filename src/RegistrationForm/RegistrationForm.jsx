import React, { useState } from "react";
import Confirmation from "../Confirmation/Confirmation.jsx";
import PropTypes from "prop-types";
import styles from "./RegistrationForm.module.css";

const RegistrationForm = (props) => {
    const [submittedForm, setSubmittedForm] = useState();
    function onSubmit(e) {
        e.preventDefault();
        const data = new FormData(e.target);
        setSubmittedForm(data);
    };

    return (
        <>
            {
                submittedForm ? (
                    <Confirmation name={submittedForm.get("myName")} contactNumber={submittedForm.get("myContactNumber")} email={submittedForm.get("myEmail")} event={props.event} />
                ) : (
                    <div className={styles.container}>
                        <div className={styles.eventDetailsContainer}>
                            <div className={styles.title}>
                                <p>Register for <span>{props.event.title}</span></p>
                            </div>
                            <div className={styles.text}>
                                <p>Event Date: <time dateTime={props.event.date}>{props.event.eventDate}</time></p>
                                <p>Event Time: <time dateTime={props.event.time}>{props.event.eventTime}</time></p>
                            </div>
                            <div className={styles.text}>
                                <p>Event Location: <span>{props.event.location}</span></p>
                            </div>
                            <div className={styles.text}>
                                <p>Organized by: <span>{props.event.organizerName}</span></p>
                            </div>
                        </div>
                        <form onSubmit={onSubmit} className={styles.registrationForm}>
                            <div className={styles.title}>
                                <p>Participant's Information: </p>
                            </div>
                            <div className={styles.formField}>
                                <label htmlFor="nameId">Name: </label>
                                <input className={styles.input} type="text" name="myName" id="nameId" required/>
                            </div>
                            
                            <div className={styles.formField}>
                                <label htmlFor="contactId">Phone Number: </label>
                                <input className={styles.input} type="tel" name="myContactNumber" id="contactId" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}" placeholder="123-456-7890" required/>
                            </div>
                            
                            <div className={styles.formField}>
                                <label htmlFor="emailId">Email: </label>
                                <input className={styles.input} type="email" name="myEmail" id="emailId" placeholder="sou@example.com" required/>
                            </div>
                            
                            <div className={styles.buttonContainer}>
                                <input className={styles.submitButton} type="submit" value="Register"/>
                            </div>
                        </form>
                    </div>
                )
            }
        </>
    );
};

RegistrationForm.propTypes = {
    event: PropTypes.object.isRequired,
};

export default RegistrationForm;