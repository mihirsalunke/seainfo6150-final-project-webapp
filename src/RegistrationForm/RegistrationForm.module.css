@import url("https://fonts.googleapis.com/css2?family=Rowdies:wght@300&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Nerko+One&display=swap");

.container {
    background-image: linear-gradient(to right, rgb(160, 201, 226), rgb(207, 233, 248) 70%);
    max-width: 700px;
    margin: 2rem auto;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.3);
    justify-content: center;
    border-radius: 10px;
}

.eventDetailsContainer {
    margin: 2em 5em;
    
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.title {
    text-align: center;
    font-family: Rowdies, sans-serif;
    font-size: 2em;
}

.text {
    padding-left: 5%;
    font-size: larger;
}

.text span {
    font-family: "Nerko One", sans-serif;
}

.text time {
    font-family: "Nerko One", sans-serif;
}

.registrationForm {
    margin: 2em 5em;
    display: flex;
    flex-flow: column wrap;
    font-size: 100%;
    color:  black;
}

.formField {
    display: flex;
    justify-content: space-evenly;
}
.formField label {
    width: 40%;
    text-align: right;
}
.formField input {
    width: 40%;
}

.submitButton {
    background: orange;
    border-radius: 10px;
    color: #fff;
    padding: 10px;
    font-family: Rowdies, sans-serif;
    font-size: 1.5em;
    margin: 0 0 20px;
    width: 5em;
    text-align: center;
    vertical-align: middle;
    align-self: center;
    transition: transform .25s ease-out;
}

.submitButton:hover {
    background: green;
    transform: translateY(2px);
    cursor: pointer;
}

.buttonContainer {
    margin-top: 1em;
    align-self: center;
}

.submitButton:active {
    transform: translateY(1px);
}