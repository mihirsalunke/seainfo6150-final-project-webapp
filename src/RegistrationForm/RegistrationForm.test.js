import React from "react";
import RegistrationForm from "./RegistrationForm.jsx";
import { render } from "@testing-library/react";

const event = {
    "eventId": "1110001",
    "title": "Future of AI",
    "description": "Future of AI is showcasing latest innovation and business modules",
    "image" : "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS9tSKBQBaTBTW3fMOh_8KPz5CqGeT6AeptUA&usqp=CAU",
    "location": "Chicago",
    "organizerName": "B2C Tech Inc",
    "organizerContactNumber": "351-800-0120",
    "category": "Tech",
    "date": "2020-12-24",
    "time": "13:00",
    "eventDate": "December 24th 2020",
    "eventTime": "1:00 pm"
};

describe("RegistrationForm tests", () => {
    it("renders correctly", () => {
      const { container } = render(<RegistrationForm event={event} />);
      expect(container).toMatchSnapshot();
    });
});