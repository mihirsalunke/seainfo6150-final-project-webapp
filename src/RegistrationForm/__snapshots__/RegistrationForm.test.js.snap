// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="eventDetailsContainer"
    >
      <div
        class="title"
      >
        <p>
          Register for 
          <span>
            Future of AI
          </span>
        </p>
      </div>
      <div
        class="text"
      >
        <p>
          Event Date: 
          <time
            datetime="2020-12-24"
          >
            December 24th 2020
          </time>
        </p>
        <p>
          Event Time: 
          <time
            datetime="13:00"
          >
            1:00 pm
          </time>
        </p>
      </div>
      <div
        class="text"
      >
        <p>
          Event Location: 
          <span>
            Chicago
          </span>
        </p>
      </div>
      <div
        class="text"
      >
        <p>
          Organized by: 
          <span>
            B2C Tech Inc
          </span>
        </p>
      </div>
    </div>
    <form
      class="registrationForm"
    >
      <div
        class="title"
      >
        <p>
          Participant's Information: 
        </p>
      </div>
      <div
        class="formField"
      >
        <label
          for="nameId"
        >
          Name: 
        </label>
        <input
          class="input"
          id="nameId"
          name="myName"
          required=""
          type="text"
        />
      </div>
      <div
        class="formField"
      >
        <label
          for="contactId"
        >
          Phone Number: 
        </label>
        <input
          class="input"
          id="contactId"
          name="myContactNumber"
          pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
          placeholder="123-456-7890"
          required=""
          type="tel"
        />
      </div>
      <div
        class="formField"
      >
        <label
          for="emailId"
        >
          Email: 
        </label>
        <input
          class="input"
          id="emailId"
          name="myEmail"
          placeholder="sou@example.com"
          required=""
          type="email"
        />
      </div>
      <div
        class="buttonContainer"
      >
        <input
          class="submitButton"
          type="submit"
          value="Register"
        />
      </div>
    </form>
  </div>
</div>
`;
